server/service/base_client.go

```diff
@@ -65,7 +75,49 @@ func (bc *baseClient) url(path, rawQuery string) *url.URL {
 	return &u
 }
 
-func newBaseClient(addr string, insecureSkipVerify bool, rootCA, urlPrefix string) (*baseClient, error) {
+// setServerCapabilities updates the server capabilities based on the response
+// from the server.
+func (bc *baseClient) setServerCapabilities(response *http.Response) {
+	bc.serverCapabilities = map[fleet.Capability]struct{}{}
+	capabilities := response.Header.Get("X-Fleet-Capabilities")
+
+	if capabilities == "" {
+		return
+	}
+
+	for _, capability := range strings.Split(capabilities, ",") {
+		bc.serverCapabilities[fleet.Capability(capability)] = struct{}{}
+	}
+}
+
+// HasServerCapability returns a boolean indicating if the server supports the
+// given capability
+func (bc *baseClient) HasServerCapability(capability fleet.Capability) bool {
+	_, ok := bc.serverCapabilities[capability]
+	return ok
+}
+
+// setClientCapabilities header is used to set a header with the client
+// capabilities in the given request.
+//
+// This method is defined in baseClient because other clients generally have
+// custom implementations of a method to perform the requests to the server.
+func (bc *baseClient) setClientCapabilitiesHeader(req *http.Request) {
+	if len(bc.clientCapabilities) == 0 {
+		return
+	}
+	capabilities := make([]string, len(bc.clientCapabilities))
```

Comment:
Nit: See prev comment about making `bc.clientCapabilities` a map. I don't think is a huge deal, but the current code doesn't take into account the possibility of having dups in `bc.clientCapabilities`